apps/user-office-backend/src/datasources/postgres/ProposalSettingsDataSource.ts

```diff
@@ -540,4 +544,76 @@ export default class PostgresProposalSettingsDataSource
         );
       });
   }
+
+  // TODO: This might need to be moved to it's own ProposalStatusActionsDataSource.ts file
+  private createProposalStatusActionObject(
+    proposalActionStatusRecord: ProposalStatusActionRecord &
+      ProposalWorkflowConnectionHasActionsRecord & {
+        config: ProposalStatusActionConfigBase;
+      }
+  ) {
+    return new ProposalStatusAction(
+      proposalActionStatusRecord.proposal_status_action_id,
+      proposalActionStatusRecord.connection_id,
+      proposalActionStatusRecord.name,
+      proposalActionStatusRecord.type,
+      proposalActionStatusRecord.executed,
+      proposalActionStatusRecord.config
+    );
+  }
+
+  async getStatusActionsByConnectionId(
+    proposalWorkflowConnectionId: number,
+    proposalWorkflowId: number
+  ): Promise<ProposalStatusAction[]> {
+    const proposalActionStatusRecords: (ProposalStatusActionRecord &
+      ProposalWorkflowConnectionHasActionsRecord & {
+        config: ProposalStatusActionConfigBase;
+      })[] = await database
+      .select()
+      .from('proposal_status_actions as psa')
+      .join('proposal_workflow_connection_has_actions as pwca', {
+        'pwca.action_id': 'psa.proposal_status_action_id',
+      })
+      .where('pwca.workflow_id', proposalWorkflowId)
+      .andWhere('pwca.connection_id', proposalWorkflowConnectionId);
+
+    const proposalStatusActions = proposalActionStatusRecords.map(
+      (proposalActionStatusRecord) =>
+        this.createProposalStatusActionObject(proposalActionStatusRecord)
+    );
+
+    return proposalStatusActions;
+  }
+
+  async updateStatusAction(

```

Comment:
Suggestion: This feels a little counterintuitive that the parameter accepts all object properties, including `name`, but updates only `executed` and `config`. Maybe you could use the Pick modifier or alternatively update all fields

Reply:
I think the other properties are not updated but they are still used in the `where` clause and in the return result.